### PR TITLE
Use staging-csi to work around quay.io availability

### LINKED ISSUE #91242: [Failing Test] Storage tests are failing due to quay.io being down
**Which jobs are failing**:

`skew-cluster-latest-kubectl-stable1-gce`
`gce-cos-master-default`
`gce-cos-master-alpha-features`
`gci-gce-ingress`
`gce-cos-master-reboot`
`gce-cos-master-scalability-100`
`Conformance - GCE - Master`
`integration-master`
`kind-master-parallel`
`kind-ipv6-master-parallel`

**Which test(s) are failing**:

All relying on volume snapshot controller.

**Since when has it been failing**:

`5/18/20 12:38 AM PT`

**Testgrid link**:

https://testgrid.k8s.io/sig-release-master-blocking

**Reason for failure**:

https://quay.io/ is experiencing downtime, causing failure to pull image in volume snapshot controller deployment: https://github.com/kubernetes/kubernetes/blob/6a0e01880ad7ea4c110040c3be417516c0bf6bca/cluster/addons/volumesnapshots/volume-snapshot-controller/volume-snapshot-controller-deployment.yaml#L26

**Anything else we need to know**:

/cc @kubernetes/ci-signal
/milestone v1.19
/priority critical-urgent
/sig testing
/sig storage

### PR DESCRIPTION
**Pull request by dims**
Quay.io is down, let's move things over to staging-csi and use those images. Here's the script i used to copy things over:
https://gist.github.com/dims/0aef844c22af80ed3637745071c8854c

Signed-off-by: Davanum Srinivas <davanum@gmail.com>

**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #91242  (Short term fix! longer term fix is for CSI folks to publish to staging-csi using automation and promote to prod using image-promoter)

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm

Thank you @dims!

@kubernetes/ci-signal

**Comment by BenTheElder** [comment]
> May 19 16:11:36.747: FAIL: deploying csi-hostpath driver: Registry: gcr.io/k8s-staging-csi is missing in test/utils/image/manifest.go, please add the registry, otherwise the test will fail on air-gapped clusters

**Comment by dims** [comment]
@BenTheElder looks like we have one more `quay.io/kubernetes_incubator/nfs-provisioner:v2.2.2` to deal with

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -226,7 +223,7 @@ func initImageConfigs() map[int]Config {
 	configs[JessieDnsutils] = Config{e2eRegistry, "jessie-dnsutils", "1.0"}
 	configs[Kitten] = Config{e2eRegistry, "kitten", "1.0"}
 	configs[Nautilus] = Config{e2eRegistry, "nautilus", "1.0"}
-	configs[NFSProvisioner] = Config{quayIncubator, "nfs-provisioner", "v2.2.2"}
+	configs[NFSProvisioner] = Config{k8sCSI, "nfs-provisioner", "v2.2.2"}
```
Comment (msau42) [inline test/utils/image/manifest.go:226]: nfs-provisioner is technically not part of the kubernetes-csi project, but we can put it here temporarily just for unblocking pull jobs. As part of moving the nfs provisioners to kubernetes-sigs, we'll also need to revamp the build and release process for them. cc @kmova

Comment (dims) [reply test/utils/image/manifest.go:226]: Agreed. thanks @msau42

**Comment by BenTheElder** [comment]
/lgtm
sorry was out on an errand. 
hopefully we can get this in soon ...

**Pull request merged**